tsl/test/sql/compression_update_delete.sql

```diff
@@ -731,3 +731,145 @@ SELECT COUNT(*) FROM :COMPRESS_CHUNK_1;
 SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE c4 IS NULL;
 
 DROP TABLE sample_table;
+
+-- test filtering with ORDER BY columns
+CREATE TABLE sample_table(time timestamptz, c1 int, c2 int, c3 int, c4 int);
+SELECT create_hypertable('sample_table','time');
+ALTER TABLE sample_table SET (timescaledb.compress,timescaledb.compress_segmentby='c4', timescaledb.compress_orderby='c1,c2,time');
+INSERT INTO sample_table
+SELECT t, c1, c2, c3, c4
+FROM generate_series(:'start_date'::timestamptz - INTERVAL '9 hours',
+            :'start_date'::timestamptz,
+            INTERVAL '1 hour') t,
+    generate_series(0,9,1) c1,
+    generate_series(0,9,1) c2,
+    generate_series(0,9,1) c3,
+    generate_series(0,9,1) c4;
+SELECT compress_chunk(show_chunks('sample_table'));
+
+-- get FIRST chunk
+SELECT ch1.schema_name|| '.' || ch1.table_name AS "CHUNK_1"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.hypertable ht
+WHERE ch1.hypertable_id = ht.id AND ch1.table_name LIKE '_hyper_%'
+ORDER BY ch1.id \gset
+
+-- get FIRST compressed chunk
+SELECT ch1.schema_name|| '.' || ch1.table_name AS "COMPRESS_CHUNK_1"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.hypertable ht
+WHERE ch1.hypertable_id = ht.id AND ch1.table_name LIKE 'compress_%'
+ORDER BY ch1.id \gset
+
+-- check that you uncompress and delete only for exact SEGMENTBY value
+BEGIN;
```

Comment:
I think it would be more clear to also validate rows present in compressed/uncompressed chunk before and after UPDATE/DELETE. This will record the actual filtering happening as part of the PR.

ex:
BEGIN;
-- total rows in compressed chunk
SELECT COUNT(*) as "total_rows" FROM :COMPRESS_CHUNK_1 \gset
-- total rows in compressed chunk matching condition c4 = 5
SELECT COUNT(*) as "total_affected_rows" FROM :COMPRESS_CHUNK_1 where c4 = 5  \gset
DELETE FROM sample_table WHERE c4 = 5;
-- report true
SELECT COUNT(*) = ((:total_rows) - (:total_affected_rows)) FROM :COMPRESS_CHUNK_1;
ROLLBACK;